### PR TITLE
Named certificates

### PR DESCRIPTION
**Pull request by brucejcong**
I ran`hypershift render` with the cluster.yaml file set up with named certs along with the pull-secret.txt file.
```
namedCerts:
  - namedCertPrefix: cert1
    namedCertDomain: c1.hosted.example.com
  - namedCertPrefix: cert2
    namedCertDomain: c2.hosted.example.com```

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
+	IngressSubdomain        string       `json:"ingressSubdomain"`
+	OpenShiftAPIClusterIP   string       `json:"openshiftAPIClusterIP"`
+	ImageRegistryHTTPSecret string       `json:"imageRegistryHTTPSecret"`
+	RouterNodePortHTTP      string       `json:"routerNodePortHTTP"`
+	RouterNodePortHTTPS     string       `json:"routerNodePortHTTPS"`
+	OpenVPNNodePort         string       `json:"openVPNNodePort"`
+	BaseDomain              string       `json:"baseDomain"`
 }
+
+type NamedCerts struct {
```
Comment (csrwng) [inline pkg/api/types.go]: Please rename to `NamedCert`

**Review by csrwng — COMMENTED** [review]
@brucejcong looks good, just one nit

File: cluster.yaml.example [hunk 2]
```diff
@@ -5,6 +5,11 @@ externalAPIAddress: 10.0.0.1
 externalVPNDNSName: vpn.hosted.example.com
 externalVPNPort: 3118
 serviceCIDR: 172.31.0.0/16
+namedCerts:
+  - namedCertPrefix: cert1
```
Comment (relyt0925) [inline cluster.yaml.example]: make the prefix include the path as well it needs to be the whole path
`"/etc/kubernetes/secret/cert1.crt"`

Comment (relyt0925) [reply cluster.yaml.example]: it should ultimately be `/etc/kubernetes/secret/cert1` since that is where we will put these named certs for now

**Pull request merged**